src/Illuminate/Foundation/Http/FormRequest.php

```diff
@@ -110,11 +112,27 @@ protected function getValidatorInstance()
     protected function createDefaultValidator(ValidationFactory $factory)
     {
         return $factory->make(
-            $this->validationData(), $this->container->call([$this, 'rules']),
+            $this->validationData(), $this->resolveRules(),
             $this->messages(), $this->attributes()
         )->stopOnFirstFailure($this->stopOnFirstFailure);
     }
 
+    /**
+     * The rules to validate against.
+     *
+     * @return array
+     */
+    protected function resolveRules()
+    {
+        if (! $this->precognitive() || ! $this->headers->has('Precognition-Validate-Only')) {
+            return $this->container->call([$this, 'rules']);
+        }
+
+        return Collection::make($this->container->call([$this, 'rules']))
+            ->only(explode(',', $this->header('Precognition-Validate-Only')))
```

Comment:
Are we being strict on saying that the header must be comma separated **without** spaces?

```
first_name,email
```

Currently, this would not validate:

```
first_name, email
```

Since we have a collection we could trim spaces from each item.

Reply:
I did consider doing this, however it is valid (not saying it is good!) to surround input name with spaces:

```
POST

Content-type application/json

{" input with spaces ":"expected-value"}
```

```php
dd(request()->input(' input with spaces ');
```

<img width="636" alt="Screen Shot 2022-08-19 at 12 14 45 pm" src="https://user-images.githubusercontent.com/24803032/185527805-87ec5de8-770c-4234-9d81-fec7c81cc1ea.png">

So if we do this, there is potential it will cause issues or not be supported by some cases.

I'm happy to exclude those cases - but as long as we do it intentionally.

Reply:
Hmm, tricky one! But yeah, if it's valid within the `Request` class, we should keep it valid here.

Reply:
bit of a pain really. isn't valid for traditional forms, but totally valid for JSON.